File: FoodOrderingSystem/frontend/src/api/orderService.js
import { orderApi } from "./axiosConfig";

export const createOrder = (orderPayload) => orderApi.post("/orders", orderPayload);

export const getAllOrders = () => orderApi.get("/orders");

export const getOrderById = (id) => orderApi.get(`/orders/${id}`);

export const updateOrder = (id, orderPayload) => orderApi.put(`/orders/${id}`, orderPayload);

export const deleteOrder = (id) => orderApi.delete(`/orders/${id}`);

export const getOrdersByStatus = (status) => orderApi.get(`/orders/status/${status}`);

export const getOrdersByCustomer = (customerName) =>
    orderApi.get(`/orders/customer/${encodeURIComponent(customerName)}`);

export const getOrdersByFood = (foodItem) =>
    orderApi.get(`/orders/food/${encodeURIComponent(foodItem)}`);

File: FoodOrderingSystem/order-service/src/main/java/com/foodorder/orderservice/service/OrderService.java
package com.foodorder.orderservice.service;

import com.foodorder.orderservice.dto.OrderRequestDTO;
import com.foodorder.orderservice.dto.OrderResponseDTO;

import java.util.List;

public interface OrderService {

    OrderResponseDTO createOrder(OrderRequestDTO requestDTO);

    List<OrderResponseDTO> getAllOrders();

    OrderResponseDTO getOrderById(Long id);

    OrderResponseDTO updateOrder(Long id, OrderRequestDTO requestDTO);

    void deleteOrder(Long id);

    List<OrderResponseDTO> getOrdersByStatus(String status);

    List<OrderResponseDTO> getOrdersByCustomer(String customerName);

    List<OrderResponseDTO> getOrdersByFood(String foodItem);

    OrderResponseDTO updateOrderStatus(Long id, String status);
}
